### PR TITLE
[0.7] Update for Laravel 5.x

### PR DESCRIPTION
**Pull request by barryvdh**
Changes:

 - [x] Update constraints
 - [x] New config system (no namespaces, load manually, move out src folder)
 - [x] Html + Form are not included by default
 - [ ] Fix the tests
 - [ ] More changes?

### CODE REVIEW

**Comment by barryvdh** [comment]
I can confirm that at least basic templates work with a fresh 5 install. Haven't tested further yet.

**Comment by RyanThompson** [comment]
All this jazz works perfectly as well as custom extensions: https://github.com/anomalylabs/streams-platform/tree/master/resources/views

I'd say she's good to go.

**Comment by barryvdh** [comment]
Yeah except the tests ;)
But you can just require 0.7@dev for the time being.

**Comment by RyanThompson** [comment]
Already have! Turns out I am medically stupid when it comes to tests. Still learning ^_^

File: README.md [hunk 1]
```diff
@@ -8,28 +8,21 @@ Allows you to use [Twig](http://twig.sensiolabs.org/) seamlessly in [Laravel 4](
 
 # Requirements
 
-TwigBridge >=0.6 requires PHP 5.4+ & Laravel 4.1+.
+TwigBridge >=0.7 requires PHP 5.4+ & Laravel 5+.
```
Comment (GrahamCampbell) [inline README.md]: surely this should say "Laravel 5", not "Laravel 5+`?

File: README.md [hunk 2]
```diff
@@ -8,28 +8,21 @@ Allows you to use [Twig](http://twig.sensiolabs.org/) seamlessly in [Laravel 4](
...
+Require this package with Composer
 
-```javascript
-{
-    "require": {
-        "rcrowe/twigbridge": "0.6.*"
-    }
-}
-```
+    composer require rcrowe/twigbridge
```
Comment (GrahamCampbell) [inline README.md]: maybe use bash syntax highlighting here.

File: src/ServiceProvider.php [hunk 3]
```diff
@@ -91,11 +92,11 @@ protected function registerCommands()
     protected function registerOptions()
     {
         $this->app->bindIf('twig.extension', function () {
-            return $this->app['config']->get('twigbridge::twig.extension');
```
Comment (GrahamCampbell) [inline src/ServiceProvider.php]: Why have you removed the `::` stuff?

File: README.md (hunk 2 above)
Comment (RyanThompson) [inline README.md]: For what it's worth I think the require "rcrowe/twigbridge": "0.6.*" part is nice to see too - I, for one, build all my composer files by hand and it's nice to copy / paste it.

File: src/ServiceProvider.php (hunk 3 above)
Comment (barryvdh) [inline src/ServiceProvider.php]: Laravel 5 doesnt really do namespaces anymore right?

File: README.md (hunk 2 above)
Comment (barryvdh) [inline README.md]: Meh, the command gives you always the most recent version.

File: src/ServiceProvider.php (hunk 3 above)
Comment (RyanThompson) [inline src/ServiceProvider.php]: Namespaces are definitely still around just it's more up to us as the outside packages. If not using the class path as an abstract like this I really like using a hard separator for the namespace. Personally, I think it reads better.

Comment (GrahamCampbell) [inline src/ServiceProvider.php]: That doesn't mean you can't format the keys in the same way. It still supports view namespaces, so keeping the `::` feels consistent.

Comment (GrahamCampbell) [inline src/ServiceProvider.php]: > Namespaces are definitely still around

No, namespaces were removed from the config compoent.

File: README.md (hunk 2 above)
Comment (RyanThompson) [inline README.md]: Lang too ^_^


Ryan Thompson
President / CEO
AnomalyLabs, Inc.
www.anomaly.is

t (877) 705-0806
m (309) 752-8581

File: src/ServiceProvider.php (hunk 3 above)
Comment (RyanThompson) [inline src/ServiceProvider.php]: Lang uses em too. I don't think in removing the addNamespace from config he meant to discourage using them in config keys though. It was more of a loader / performance change as I see it. Config / Lang / View namespaces I think are all still good practice.

Comment (barryvdh) [inline src/ServiceProvider.php]: https://mobile.twitter.com/taylorotwell/status/553646428296593408

> Taylor: I would just do package.key

Comment (GrahamCampbell) [inline src/ServiceProvider.php]: Ok. I'm not sure what people are going to adopt. I was going to keep the namespaces, but I might switch it all in my packges.

Comment (RyanThompson) [inline src/ServiceProvider.php]: Awesome - good to know!

**Comment by barryvdh** [comment]
Done.

I've used `$this->app->alias('twig', 'TwigBridge\Bridge');`

Just thinking if it would make sense to alias the Twigbridge to Twig_Environment also, because that's what you would normally use..

**Comment by RyanThompson** [comment]
Killer!


Ryan Thompson
President / CEO
AnomalyLabs, Inc.
www.anomaly.is

t (877) 705-0806
m (309) 752-8581

File: src/ServiceProvider.php (hunk 3 above)
Comment (rcrowe) [inline src/ServiceProvider.php]: Agreed, lets dump the use of `::`

**Comment by rcrowe** [comment]
I think that seems like a good thing todo @barryvdh aliasing `Twig_Environment`.

**Comment by barryvdh** [comment]
Okay did that, also moved the register methods to register.

**Comment by barryvdh** [comment]
So, should we branch the current master to a 0.6 branch and merge this in?

**Comment by bcho** [comment]
@barryvdh Please publish 0.7!

**Comment by barryvdh** [comment]
You can still use it already.

**Comment by bcho** [comment]
@barryvdh merge 0.7 into master branch =D

**Comment by barryvdh** [comment]
I've tagged a release, so that's one thing ;)

**Comment by barryvdh** [comment]
Bit late, but it's merged now :)

**Pull request merged**